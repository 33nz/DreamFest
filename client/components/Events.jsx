import React from 'react'
import { Link } from 'react-router-dom'
import EventItem from './EventItem'

const events = [{ title: 'Weeds worker bee', datetime: '28/09/2020 8.00 am', volunteersNeeded: '8', description: 'We need to weed the upper planting beds' }]

function Events () {
  return (
    <div className="m-6">
    <h2 className="my-4 is-pulled-left">Events</h2>
    <Link to='/events/new' className="button my-4 is-primary is-pulled-right">Add new event</Link>
    <div className="box clearfix">
      {events.map((event, index) =>
        <EventItem key={index} event={event} />
      )}
    </div>
     </div>
  )
}

export default Events

// component did mount = useEffect 
// - function API call to get DB info