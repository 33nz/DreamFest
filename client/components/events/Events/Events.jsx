import React from 'react'
import { Link } from 'react-router-dom'
import { useSelector } from 'react-redux'

import EventItem from '../EventItem/EventItem'

export default function Events ({ events }) {
  const isAdmin = useSelector(globalState => globalState.user.isAdmin)

  return (
    <>
      <h1 className='events-title'>Events</h1>
      <section>
        {events.map((event) =>
          <EventItem key={event.id} event={event} isAdmin={isAdmin} />
        )}
      </section>
      <div>
        {
          isAdmin
            ? <Link to='/event/new' className='button'>Add New Event</Link>
            : null
        }
      </div>
    </>
  )
}
