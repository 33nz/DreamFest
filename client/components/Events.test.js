import React from 'react'
import { HashRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

import Events from './Events'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <HashRouter>
        <Events />
      </HashRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
