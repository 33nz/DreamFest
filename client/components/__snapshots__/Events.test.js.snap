// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h2
    className="my-4"
  >
    Events
  </h2>,
  <div
    className="box"
  >
    <div>
      <h4>
        Weeds worker bee
      </h4>
      <p>
        28/09/2020 8.00 am
      </p>
      <p>
        8
         volunteers needed
      </p>
      <p>
        We need to weed the upper planting beds
      </p>
    </div>
  </div>,
]
`;
